code/datums/status_effects/buffs/stun_absorption.dm

```diff
@@ -0,0 +1,236 @@
+
+/**
+ * Signal proc for generic stun signals being sent, such as [COMSIG_LIVING_STATUS_STUN] or [COMSIG_LIVING_STATUS_KNOCKDOWN].
+ *
+ * When we get stunned, we will try to absorb a number of seconds from the stun, and return [COMPONENT_NO_STUN] if we succeed.
+ */
+/datum/status_effect/stun_absorption/proc/try_absorb_incapacitating_effect(mob/living/source, amount = 0, ignore_canstun = FALSE)
+	SIGNAL_HANDLER
+
+	// we blocked a stun this tick that resulting is us qdeling, so stop
+	if(QDELING(src))
+		return NONE
+
+	// Amount less than (or equal to) zero is removing stuns, so we don't want to block that
+	if(amount <= 0 || ignore_canstun)
+		return NONE
+
+	if(!absorb_stun(amount))
+		return NONE
+
+	return COMPONENT_NO_STUN
+
+/**
+ * Signal proc for [COMSIG_LIVING_GENERIC_STUN_CHECK]. (Note, this includes being stamcrit)
+ *
+ * Whenever a generic stun check is done against us, we'll just try to block it with "0 second" stun.
+ * This prevents spam us from showing feedback messages, and is for the generic "can be stunned" check.
+ */
+/datum/status_effect/stun_absorption/proc/try_absorb_generic_effect(mob/living/source, check_flags)
+	SIGNAL_HANDLER
+
+	if(QDELING(src))
+		return NONE
+
+	// "0 amount" / "0 seconds of stun" is used so no feedback is sent on success
+	if(!absorb_stun(0))
+		return NONE
+
+	return COMPONENT_NO_STUN
+
+/**
+ * Absorb a number of seconds of stuns.
+ * If we hit the max amount of absorption, we will qdel ourself in this proc.
+ *
+ * * amount - this is the number of deciseconds being absorbed at once.
+ *
+ * Returns TRUE on successful absorption, or FALSE otherwise.
+ */
+/datum/status_effect/stun_absorption/proc/absorb_stun(amount)
+	if(owner.stat != CONSCIOUS)
+		return FALSE
+
+	// Now we gotta check that no other stun absorption we have is blocking us
+	for(var/datum/status_effect/stun_absorption/similar_effect in owner.status_effects)
+		if(similar_effect == src)
+			continue
+		// they blocked a stun this tick that resulted in them qdeling, so disregard
+		if(QDELING(similar_effect))
+			continue
+		// if we have another stun absorption with higher priority,
+		// don't do anything, let them handle it instead
+		if(similar_effect.priority > priority)
+			return FALSE
+
+	// At this point, a stun was successfully absorbed
+	. = TRUE
+
+	// Only do effects if the amount was > 0 seconds
+	if(amount > 0 SECONDS)
+		// Show the message
+		if(shown_message)
+			// We do this replacement meme, instead of just setting it up in creation,
+			// so that we respect indentity changes done while active
+			var/really_shown_message = replacetext(shown_message, "%EFFECT_OWNER", "[owner]")
+			owner.visible_message(really_shown_message, ignored_mobs = owner)
+
+		// Send the self message
+		if(self_message)
+			to_chat(owner, self_message)
+
+		// Count seconds absorbed
+		seconds_of_stuns_absorbed += amount
+		if(seconds_of_stuns_absorbed >= max_seconds_of_stuns_blocked)
+			qdel(src)
+
+	return .
+
```

Comment:
shouldn't we just get rid of the funny dot operator and just have it `return TRUE` at the end of this proc? This seems really hardy against runtimes so I don't really get why we need it.

Reply:
like we literally do `return FALSE` everywhere else in this proc it's just plumb confusing